### PR TITLE
Refactor fty::defset implementation so it works with fty::deftypes

### PR DESCRIPTION
**Pull request by westfold**

### CODE REVIEW

**Comment by airbornemihir** [comment]
Hi Stephen - I thought I'd help by changing the branch your pull request would be merged into to testing, the usual branch. If you have reasons for merging into master, by all means change it back.

**Comment by westfold** [comment]
Thanks Mihir. I had intended to merge it into kestrel-testing, but testing should also be fine.

Stephen

File: books/centaur/fty/deftypes.lisp [hunk 1]
```diff
@@ -1053,3 +1056,24 @@
 (defmacro defprod (&whole form &rest args)
   (declare (ignore args))
   `(make-event (defprod-fn ',form state)))
+
+
+(defun defset-fn (whole state)
+  (b* ((our-fixtypes (list (flextype-form->fixtype whole)))
+       (fixtype-al (append our-fixtypes
+                           (get-fixtypes-alist (w state))))
+       (x (parse-flexlist (cdr whole) nil our-fixtypes fixtype-al state))
```
Comment (solswords) [inline books/centaur/fty/deftypes.lisp:1065]: Should this be a call of parse-flexset?

Comment (solswords) [reply books/centaur/fty/deftypes.lisp:1065]: (It's in the definition of defset-fn, if that's not clear.)

**Comment by westfold** [comment]
Yes. Clearly I need to augment the tests for non-recursive uses of fty::defset.

Thanks,
Stephen

**Comment by westfold** [comment]
This is fixed now. The error seemed to only affect error detection and reporting as the defset options are a subset of the deflist ones.

Stephen

**Pull request merged**